components-core/src/main/java/com/adyen/checkout/components/api/LogoConnectionTask.kt

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2019 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by caiof on 11/3/2019.
+ */
+package com.adyen.checkout.components.api
+
+import android.graphics.drawable.BitmapDrawable
+import com.adyen.checkout.core.api.ConnectionTask
+import com.adyen.checkout.core.api.ThreadManager
+import com.adyen.checkout.core.log.LogUtil
+import com.adyen.checkout.core.log.Logger
+import java.util.ArrayList
+import java.util.concurrent.ExecutionException
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.TimeoutException
+
+/**
+ * Task that wraps a Connection to get a Logo.
+ */
+class LogoConnectionTask(
+    private val logoApi: LogoApi,
+    val logoUrl: String,
+    callback: LogoCallback
+) : ConnectionTask<BitmapDrawable>(
+    LogoConnection(logoUrl)
+) {
+
+    var mCallbacks: MutableList<LogoCallback> = ArrayList()
+
+    init {
+        mCallbacks.add(callback)
+    }
+
+    override fun done() {
+        Logger.v(TAG, "done")
+        if (isCancelled) {
+            Logger.d(TAG, "canceled")
+            notifyFailed()
+            return
+        }
+        try {
+            // timeout just to make sure we don't get stuck, get call is blocking but should be finished or canceled by now.
+            val result = get(SAFETY_TIMEOUT.toLong(), TimeUnit.MILLISECONDS)
+            notifyLogo(result)
+        } catch (e: ExecutionException) {
+            Logger.e(TAG, "Execution failed for logo  - $logoUrl")
+            notifyFailed()
+        } catch (e: InterruptedException) {
+            Logger.e(TAG, "Execution interrupted.", e)
+            notifyFailed()
+        } catch (e: TimeoutException) {
+            Logger.e(TAG, "Execution timed out.", e)
+            notifyFailed()
+        }
+    }
+
+    fun addCallback(callback: LogoCallback) {
```

Comment:
I would `synchronize` the content of `addCallback`, `notifyCallbacksReceived` and `notifyCallbacksFailed`  just to make sure there are not called at the same time.